_pages/documentation/general_docs/kconfig_build_system.md

```diff
@@ -0,0 +1,203 @@
+---
+layout: documentation
+title: Kconfig Build System
+doc: gem5 documentation
+parent: kconfig_build_system
+permalink: /documentation/general_docs/kconfig_build_system/
+---
+
+This guide is intended for advanced users who need to build gem5 with
+multiple ISAs or customize the build options, such as the Ruby memory protocol.
+Familiarity with the Kconfig system is required.
+
+## Build GEM5 with Kconfig Build System
+
+```bash
+scons [OPTIONS] Kconfig_command TARGET
+```
+
+Supported Kconfig commands include
+
+- defconfig
+- setconfig
+- menuconfig
+- guiconfig
+- listnewconfig
+- oldconfig
+- olddefconfig
+- savedefconfig
+
+The most common use of options are `defconfig`, `setconfig` and `menuconfig`.
+
+Use `scons --help` to list these commands, and for additional information.
+
+To build gem5 with Kconfig, there are now two steps.
+
+Initial configuration
+
+```bash
+scons defconfig gem5_build build_opts/ALL
+```
+
+This will create a configuration for the `gem5_build` build directory based on
+`build_opts/ALL`. The configuration is stored in the
+`gem5_build/gem5.build/config`.
+
+To build a target in a configured build directory, run a scons command similar
+to before.
+
+```bash
+scons gem5_build/gem5.opt
```

Comment:
Could we include a reference to the `-j [number of threads]` option in our documentation? While I was adhering to the documentation, I initially initiated the build process using a single thread. It would be beneficial to explicitly state that users have the option to specify the number of threads for the build.